### PR TITLE
ACA-60 - Replace AAT callback urls 

### PR DESCRIPTION
**Pull request by sullivangareth**
**Before creating a pull request make sure that:**

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ACA-60

### Change description ###

When running FTA locally - replace AAT callback urls with the hostname and port of the docker container running ccd-test-stubs-service

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/JsonTransformer.java [hunk 1]
```diff
@@ -73,6 +74,12 @@ public void parseDefinitionJson(String path) {
                 String sheet = null;
                 try {
                     JsonNode rootSheetArray = objectMapper.readTree(jsonFile);
+
+                    if (BeftaMain.getConfig().getTestUrl().contains("localhost")) {
+                        String rootSheetArrayString = rootSheetArray.toString().replaceAll("ccd-test-stubs-service-aat.service.core-compute-aat.internal", "ccd-test-stubs-service:5555");
+                        rootSheetArray = objectMapper.readTree(rootSheetArrayString);
+                    }
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/JsonTransformer.java:81]: This is doing the trick to all the files, and perhaps is an inefficient implementation as such. Could we please apply this trick only to some specific json files corresponding to some specific worksheets in the Excel file?

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
pls action 1 comment. otherwise good.

File: src/main/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/JsonTransformer.java (hunk 1 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/JsonTransformer.java]: Addressed in https://github.com/hmcts/befta-fw/pull/93/commits/7727e368552565070660acfa0573d34543570f48

File: src/main/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/JsonTransformer.java [hunk 2]
```diff
@@ -35,6 +36,8 @@
             "WorkBasketInputFields", "WorkBasketResultFields", "Category", "Banner", "CaseType", "ComplexTypes", "EventToComplexTypes",
             "FixedLists", "Jurisdiction", "UserProfile","SearchAlias", "SearchCasesResultFields");
 
+    private static final List<String> SHEETS_FOR_URL_SUBSTITUTIONS = Arrays.asList("BEFTA_MASTER/FT_MasterCaseType/CaseEvent.json");
+
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/JsonTransformer.java:40]: better to allow substitution in the same spreadsheets of different files. not only in befta master file.

Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/JsonTransformer.java]: Fixed in https://github.com/hmcts/befta-fw/pull/93/commits/d3e7ac8024c06c6cb06ba875509748598cae534b

**Review by MSancaktutar — APPROVED** [review]
LGTM

**Review by MSancaktutar — APPROVED** [review]

**Pull request merged**